test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -6793,6 +6793,218 @@ BOOST_AUTO_TEST_CASE(cleanup_bytes_types)
+				a = 325.5;
+				b = 450.25;
+				c = 23924029.9375;
+			}
+		}
+	)";
+	compileAndRun(sourceCode, 0, "C");
+	BOOST_CHECK(callContractFunction("a()") == encodeArgs(0));
+}
+
+BOOST_AUTO_TEST_CASE(negative_exponent)
+{
+	char const* sourceCode = R"(
+		contract C {
+			function f() returns (fixed a) {
+				a = 2 ** -5;
+			}
+		}
+	)";
+	compileAndRun(sourceCode, 0, "C");
+	BOOST_CHECK(callContractFunction("a()") == encodeArgs(0));
+}
+
+BOOST_AUTO_TEST_CASE(fixed_type_arguments)
+{
+	char const* sourceCode = R"(
+		contract C {
+			function f(fixed a, fixed b) returns (fixed, fixed) {
+				return (a, b);
+			}
+		}
+	)";
+	compileAndRun(sourceCode, 0, "C");
+	BOOST_CHECK();
+}
+
+BOOST_AUTO_TEST_CASE(fixed_type_to_int)
+{
+	char const* sourceCode = R"(
+		contract C {
+			function f(fixed a) returns (uint b) {
+				b = a;
+			}
+		}
+	)";
+	compileAndRun(sourceCode, 0, "C");
+	BOOST_CHECK();
+}
+
+BOOST_AUTO_TEST_CASE(fixed_type_addition)
+{
+	char const* sourceCode = R"(
+		contract C {
+			function f(fixed a, fixed b) returns (fixed) {
+				return (a + b);
+			}
+		}
+	)";
+	compileAndRun(sourceCode, 0, "C");
+	BOOST_CHECK();
+}
+
+BOOST_AUTO_TEST_CASE(fixed_type_subtraction)
+{
+	char const* sourceCode = R"(
+		contract C {
+			function f(fixed a, fixed b) returns (fixed) {
+				return (a - b);
+			}
+		}
+	)";
+	compileAndRun(sourceCode, 0, "C");
+	BOOST_CHECK();
+}
+
+BOOST_AUTO_TEST_CASE(fixed_type_multiplication)
+{
+	char const* sourceCode = R"(
+		contract C {
+			function f(fixed a, fixed b) returns (fixed) {
+				return (a * b);
+			}
+		}
+	)";
+	compileAndRun(sourceCode, 0, "C");
+	BOOST_CHECK();
+}
+
+BOOST_AUTO_TEST_CASE(fixed_type_division)
+{
+	char const* sourceCode = R"(
+		contract C {
+			function f(fixed a, fixed b) returns (fixed) {
+				return (a / b);
+			}
+		}
+	)";
+	compileAndRun(sourceCode, 0, "C");
+	BOOST_CHECK();
+}
+
+BOOST_AUTO_TEST_CASE(fixed_type_modulus)
+{
+	char const* sourceCode = R"(
+		contract C {
+			function f() returns (fixed) {
+				fixed a = 0.42578125 % -0.4271087646484375;
+				fixed b = .5 % a;
+				return b;
+			}
+		}
+	)";
+	compileAndRun(sourceCode, 0, "C");
+	BOOST_CHECK();
+}
+
+BOOST_AUTO_TEST_CASE(fixed_type_incrementing_decrementing)
+{
+	char const* sourceCode = R"(
+		contract C {
+			function inc() returns (ufixed) {
+				ufixed a = 1.25;
+				return a++;
+			}
+			function dec() returns (ufixed) {
+				ufixed a = 4.5;
+				a--;
+			}
+		}
+	)";
+	compileAndRun(sourceCode, 0, "C");
+	BOOST_CHECK();
+}
+
+BOOST_AUTO_TEST_CASE(var_and_fixed_one_third_equivalent)
+{
+	char const* sourceCode = R"(
+		contract C {
+			function f() returns (bool) {
+				var a = 1/3;
+				ufixed0x256 b = ufixed0x256(1/3);
+				return (a == b);
+			}
+		}
+	)";
+	compileAndRun(sourceCode, 0, "C");
+	BOOST_CHECK();
+}
+
+BOOST_AUTO_TEST_CASE(for_and_while_loops_with_fixed) 
+{
+	char const* sourceCode = R"(
+		contract C {
+			function daWhile() returns (fixed) {
+				while (var i = 3.25; i < 500; i++) {
```

Comment:
For me, this code fails to compile in browser-solidity:

```
contract Test {
    function summing() constant returns (uint sum) {
        while (uint i = 0; i < 10; i++) {
            sum += i;
        }
    }
}
```

But if you replace the `while` with `for`, it compiles just fine.